[CI] PipelineFactoryTests testCreateUnsupportedFieldAccessPattern failing

**Build Scans:**
- [elasticsearch-periodic #9101 / release-tests](https://gradle-enterprise.elastic.co/s/b7pzrk2exm22m)
- [elasticsearch-periodic #9094 / release-tests](https://gradle-enterprise.elastic.co/s/gti4jsisaiadg)
- [elasticsearch-pull-request #78512 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/6vrndaobypw4i)
- [elasticsearch-pull-request #78512 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/pfkv3xubkvs4e)
- [elasticsearch-pull-request #78493 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/uw7qwews3zwnk)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.ingest.PipelineFactoryTests.testCreateUnsupportedFieldAccessPattern" -Dtests.seed=3FF77A0DD0EBD582 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=sq-AL -Dtests.timezone=Mexico/General -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.ingest.PipelineFactoryTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testCreateUnsupportedFieldAccessPattern),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: "pipeline [_id] doesn't support value of [random] for parameter [field_access_pattern]"
     but: was "pipeline [_id] doesn't support one or more provided configuration parameters [field_access_pattern]"
```

**Issue Reasons:**
- [main] 2 consecutive failures in step release-tests
- [main] 3 consecutive failures in step checkpart1_release-tests
- [main] 5 failures in test testCreateUnsupportedFieldAccessPattern (10.6% fail rate in 47 executions)
- [main] 2 failures in step release-tests (100.0% fail rate in 2 executions)
- [main] 3 failures in step checkpart1_release-tests (100.0% fail rate in 3 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (100.0% fail rate in 2 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (20.0% fail rate in 10 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -495,9 +495,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.inference.bulk.BulkInferenceExecutorTests
   method: testSuccessfulExecution
   issue: https://github.com/elastic/elasticsearch/issues/130306
-- class: org.elasticsearch.ingest.PipelineFactoryTests
-  method: testCreateUnsupportedFieldAccessPattern
-  issue: https://github.com/elastic/elasticsearch/issues/130422
 - class: org.elasticsearch.test.rest.yaml.RcsCcsCommonYamlTestSuiteIT
   method: test {p0=msearch/20_typed_keys/Multisearch test with typed_keys parameter for sampler and significant terms}
   issue: https://github.com/elastic/elasticsearch/issues/130472

--- a/server/src/test/java/org/elasticsearch/ingest/PipelineFactoryTests.java
+++ b/server/src/test/java/org/elasticsearch/ingest/PipelineFactoryTests.java
@@ -24,6 +24,7 @@ import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assume.assumeTrue;
 import static org.mockito.Mockito.mock;
 
 public class PipelineFactoryTests extends ESTestCase {
@@ -221,12 +222,15 @@ public class PipelineFactoryTests extends ESTestCase {
     }
 
     public void testCreateUnsupportedFieldAccessPattern() throws Exception {
+        assumeTrue("Test is only valid if the logs stream feature flag is enabled", DataStream.LOGS_STREAM_FEATURE_FLAG);
         Map<String, Object> processorConfig = new HashMap<>();
         processorConfig.put(ConfigurationUtils.TAG_KEY, "test-processor");
         Map<String, Object> pipelineConfig = new HashMap<>();
         pipelineConfig.put(Pipeline.DESCRIPTION_KEY, "_description");
         pipelineConfig.put(Pipeline.VERSION_KEY, versionString);
-        pipelineConfig.put(Pipeline.FIELD_ACCESS_PATTERN, "random");
+        if (DataStream.LOGS_STREAM_FEATURE_FLAG) {
+            pipelineConfig.put(Pipeline.FIELD_ACCESS_PATTERN, "random");
+        }
         if (metadata != null) {
             pipelineConfig.put(Pipeline.META_KEY, metadata);
         }

